extension/intents/nicknames/nicknames.js

```diff
@@ -114,3 +102,43 @@ intentRunner.registerIntent({
     intentRunner.unregisterPageName(name);
   },
 });
+
+intentRunner.registerIntent({
+  name: "nicknames.pause",
+  async run(context) {
+    if (context.parentRoutine === undefined) {
+      const exc = new Error("Command not available");
+      exc.displayMessage = "Command not available";
+      throw exc;
+    }
+    context.parentRoutine.stop = true;
+    browser.storage.sync.set({
```

Comment:
Hmm... this would keep it over restarts, but I'm not sure if that's necessary. But at least it should use `browser.storage.local` as it wouldn't make sense to restart from another computer/profile (so it doesn't need to be sync'd).

Reply:
Actually I think it should just be something like:

```js
// A global
let pausedRoutine = null;

...
  async run(context) {
    if (! context.routineExecutor) {
      // error
    }
    pausedRoutine = context.routineExecutor;
    context.routineExecutor.pause();
```

And then later call `pausedRoutine.unpause()` to continue.

Actually, thinking about it a little, I think `unpauseRoutine` should be an exported function from routineExecutor.js, and that module should keep track of any paused routines. Because there's the case when you pause a routine, and another routine is already paused – that actually seems somewhat likely, since it would be easy to forget a routine halfway through. Probably when starting a new routine it makes sense to note the paused routine exists.

In some future version, it probably makes sense to keep a stack of paused routines, but only when routines are called from other routines. Though it's not entirely unreasonable to imagine a routine being paused, and the user then calling another routine during the paused routine to complete the subtask, then resuming the outer routine. But... for now, just showing some messages would be enough.